File: leetcode/two-pointers/283-Move-Zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int lastZero =0;
        int n  = nums.size();
        for(int i=0;i<n;i++){
            while(lastZero<n && nums[lastZero]!=0) lastZero++;

            if(lastZero>=0 && lastZero<n){
                if(nums[i]!=0 && lastZero<i){
                    swap(nums[i],nums[lastZero]);
                }
            }

        }
    }
};

File: leetcode/two-pointers/283-Move-Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        insertPos=0
        for i in range(len(nums)):
            if nums[i]!=0:
                nums[i],nums[insertPos]=nums[insertPos],nums[i]
                insertPos+=1
        

        
        